cnxarchive/views.py

```diff
@@ -681,10 +709,46 @@ def sitemap(environ, start_response):
                     ORDER BY revised DESC LIMIT 50000""")
             res = cursor.fetchall()
             for r in res:
-                xml.add_url('http://%s/contents/%s/%s' % (hostname,r[0], r[1]),
-                            lastmod=r[2])
+                url = 'http://%s/contents/%s/%s' % (hostname, r[0], r[1])
+                if notblocked(url):
+                    xml.add_url('http://%s/contents/%s/%s' % (
+                        hostname, r[0], r[1]), lastmod=r[2])
 
     status = '200 OK'
     headers = [('Content-type', 'text/xml')]
     start_response(status, headers)
     return [xml()]
+
+
+def robots(environ, start_response):
+    """
+    Returns a robots.txt file
+    """
+    robots_dot_txt = Robots()
+
+    bot_delays = {
+        '*': '',
+        'ScoutJet': '10',
+        'Baiduspider': '10',
+        'BecomeBot': '20',
+        'Slurp': '10'
+        }
+
+    for bot in bot_delays:
+        robots_dot_txt.add_bot(bot, bot_delays[bot], PAGES_TO_BLOCK)
+
+    status = '200 OK'
+    gmt = timezone('GMT')
+    cur_time = datetime(1, 2, 3)
```

Comment:
This line is not necessary, you should be able to do `datetime.now(gmt)` on the next line.